smalltalksrc/VMMaker/StackInterpreter.class.st

```diff
@@ -4188,14 +4188,15 @@ StackInterpreter >> checkOkayStackPage: thePage [
 { #category : 'debug support' }
 StackInterpreter >> checkOkayStackZone: writeBack [
 	"Check that all objects in the stack zone are okay"
-
+	| ok |
+	<var: #thePage type: #'StackPage *'>
 	<inline: false>
 	writeBack ifTrue: [ self writeBackHeadFramePointers ].
-
+	
+	ok := true.
 	self usedStackPagesDo: [ :thePage |
-		(self checkOkayStackPage: thePage) ifFalse: [ ^ false ] ].
-
-	^ true
+		ok := ok and: [ (self checkOkayStackPage: thePage) ] ].
+	^ ok
```

Comment:
Can't we leave the early return because it uses a lazy `and:` anyway?

Reply:
the thing is that the checkOkStackPage: and child have side effects that print things...

Reply:
Sure but it's using `and:`, so it stop calling `checkOkayStackPage:` after the first time it returns false, or am I missing something?